sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamTest.scala

```diff
@@ -133,6 +133,9 @@ trait StreamTest extends QueryTest with SharedSQLContext with TimeLimits with Be
     }
 
     def apply(rows: Row*): CheckAnswerRows = CheckAnswerRows(rows, false, false)
+
+    def apply(checkFunction: Row => Unit): CheckAnswerRows =
+      CheckAnswerRows(null, false, false, checkFunction)
```

Comment:
This construction feels very forced. I wonder if we should define a new clase class for `CheckAnswer(function)`. Maybe we could have the CheckAnswer classes share a common trait to avoid needing to duplicate the setup code for checking the answers.

I don't feel strongly about this, just a thought.